modules/blast.nf

```diff
@@ -105,9 +142,46 @@ process filter_best_bitscore {
 
     output:
     tuple val(sample_id), path("${sample_id}_${db_id}_blast_best_bitscore.csv"), emit: blast_best_bitscore_csv
+    tuple val(sample_id), path("${sample_id}_${db_id}_filter_bitscore_provenance.yml"),            emit: provenance
 
     script:
     """
     filter_best_bitscore.py -i ${full_blast_report} > ${sample_id}_${db_id}_blast_best_bitscore.csv
+
+    cat <<-EOL_VERSIONS > ${sample_id}_${db_id}_filter_bitscore_provenance.yml
+    - process_name: "${task.process}-${db_id}"
+      tools:
+      - tool_name: python
+        tool_version: \$(python3 --version | cut -d' ' -f2)
+    EOL_VERSIONS
+    """
+}
+
+process build_report {
+
+    executor 'local'
+
+    publishDir "${params.outdir}/", mode: 'copy', pattern: "*report.html"
+
+    input:
+    path(collected_blast)
+    path(database_csv)
+
+    output:
+    path("summary_report.html"),        emit: report
+    path("report_provenance.yml"),      emit: provenance
+
+    script:
+    """
+    report2.py --blast ${collected_blast} --db ${database_csv} --output summary_report.html
+
+    cat <<-EOL_VERSIONS > report_provenance.yml
+    - process_name: "${task.process}"
+      tools:
+      - tool_name: python
+        tool_version: \$(python3 --version | cut -d' ' -f2)
+      - tool_name: pandas
```

Comment:
Is pandas a tool? Maybe we should think through how to describe custom scripts in the provenance schema. pandas would fit there as a library.

Reply:
Removed `pandas` from the final provenance outputs.